src/external/wannier_shared.jl

```diff
@@ -1,5 +1,85 @@
+# Shared functionality to operate with Wannier programs (Wannier.jl and Wannier90).
+# The communication happens with the Wannier90 file formats which Wannier.jl also understands.
+
 using Dates
 
+"""
+A Gaussian-shaped initial guess. Can be used as an approximation of an s- or σ-like orbital.
+"""
+struct GaussianWannierProjection
+    center::AbstractVector{Float64}
+end
+
+function (proj::GaussianWannierProjection)(basis::PlaneWaveBasis, qs)
+    # associate a center with the fourier transform of the corresponding gaussian
+    # TODO: what is the normalization here?
+
+    model = basis.model
+    map(qs) do q
+        q_cart = recip_vector_red_to_cart(model, q)
+        exp( 2π*(-im*dot(q, proj.center) - dot(q_cart, q_cart) / 4) )
+    end
+end
+
+"""
+A p-like initial guess, using the difference of two Gaussians with different centers.
+"""
+function opposite_gaussians_projection(center1::AbstractVector, center2::AbstractVector)
+    function inner(basis, qs)
+        GaussianWannierProjection(center1)(basis, qs) - GaussianWannierProjection(center2)(basis, qs)
+    end
+end
+
+@doc raw"""
+A hydrogenic initial guess.
+
+`α` is the diffusivity, ``\frac{Z}/{a}`` where ``Z`` is the atomic number and
+    ``a`` is the Bohr radius.
+"""
+struct HydrogenicWannierProjection
+    center::AbstractVector{Float64}
+    n::Integer
+    l::Integer
+    m::Integer
+    α::Real
+end
+
+function (proj::HydrogenicWannierProjection)(basis::PlaneWaveBasis, qs)
+    # TODO: Performance can probably be improved a lot here.
+    # TODO: Some of this logic could be used for the pswfc from the UPF PSPs.
+
+    # this is same as QE pw2wannier90, r = exp(x) / α
+    xmin = -6.0
+    dx = 0.025
+    rmax = 10.0
+    n_r = round(Int, (log(rmax) - xmin) / dx) + 1
+    x = range(; start=xmin, length=n_r, step=dx)
+    # rgrid depends on α
+    r = exp.(x) ./ proj.α
+    dr = r .* dx
+    R = radial_hydrogenic(r, proj.n, proj.α)
+    r2_R_dr = r.^2 .* R .* dr
+
+    model = basis.model
+
+    map(qs) do q
+        q_cart = recip_vector_red_to_cart(model, q)
+
+        qnorm = norm(q_cart)
+        radial_part = 0
```

Comment:
Not that this function is type sable at all, but since this is a common antipattern I wanted to point it out:

This is a type instability as `radial_part` is an integer, but later floats are added, initialise as `= 0.0`.